src/simsopt/geo/surfaceobjectives.py

```diff
@@ -104,17 +114,25 @@ class ToroidalFlux(Optimizable):
 
     def __init__(self, in_surface, biotsavart, idx=0, range=None, nphi=None, ntheta=None):
         if range is None:
-            range = Surface.RANGE_FIELD_PERIOD
+            if in_surface.stellsym:
+                range = Surface.RANGE_HALF_PERIOD
+            else:
+                range = Surface.RANGE_FIELD_PERIOD
         if nphi is None:
             nphi = len(in_surface.quadpoints_phi)
         if ntheta is None:
             ntheta = len(in_surface.quadpoints_theta)
-        surface = in_surface.__class__.from_nphi_ntheta(nphi=nphi, ntheta=ntheta, range=range, nfp=in_surface.nfp, stellsym=in_surface.stellsym,\
-                                                        mpol=in_surface.mpol, ntor=in_surface.ntor, dofs=in_surface.dofs)
-        self.surface = surface
+
+        if range is not None or nphi is not None or ntheta is not None:
+            surface = in_surface.__class__.from_nphi_ntheta(nphi=nphi, ntheta=ntheta, range=range, nfp=in_surface.nfp, stellsym=in_surface.stellsym, \
+                                                            mpol=in_surface.mpol, ntor=in_surface.ntor, dofs=in_surface.dofs)
+            self.surface = surface
+        else:
+            self.surface = in_surface
```

Comment:
In the 3 surface objectives, I don't think this logic makes sense. `range`, `nphi`, and `ntheta` are all ensured to be different from `None` before line 125. So lines 130-131 never execute, right? Do you want to move lines 116-124 inside the `if` statement on line 126? (Same for `Volume` and `Area`.)